### PR TITLE
Be resilient to getting busy messages back from sqlite while retrieving data.

### PR DESCRIPTION
**Pull request by CyrusNajmabadi**
Fixes https://devdiv.visualstudio.com/DevDiv/_workitems?id=452360&fullScreen=true&_a=edit

Note: this issue should happen a lot less in 15.5 as heejae made a change in https://github.com/dotnet/roslyn/pull/21532 to have separate DBs for VS and for the OOP process.  Without having two processes hitting the same DB, there should be a lot less contention on the DB and busy exceptions should be much rarer.

Also, because we enabled WAL (write ahead logging) busy errors happen far less: https://sqlite.org/wal.html

"The second advantage of WAL-mode is that writers do not block readers and readers to do not block writers. "

### CODE REVIEW

**Comment by CyrusNajmabadi** [comment]
Tagging @dotnet/roslyn-ide

**Comment by CyrusNajmabadi** [comment]
Tagging @dpoeschl

**Comment by CyrusNajmabadi** [comment]
tagging @heejaechang

File: src/Workspaces/Core/Desktop/Workspace/SQLite/SQLitePersistentStorage_StringIds.cs [hunk 1]
```diff
@@ -12,23 +12,34 @@ internal partial class SQLitePersistentStorage
...
-        private void FetchStringTable(SqlConnection connection)
+        private bool TryFetchStringTable(SqlConnection connection)
         {
-            using (var resettableStatement = connection.GetResettableStatement(
-                $@"select * from ""{StringInfoTableName}"""))
+            try
             {
-                var statement = resettableStatement.Statement;
-                while (statement.Step() == Result.ROW)
+                using (var resettableStatement = connection.GetResettableStatement(
```
Comment (heejaechang) [inline src/Workspaces/Core/Desktop/Workspace/SQLite/SQLitePersistentStorage_StringIds.cs]: so issue was we not releasing this resource?

Comment (CyrusNajmabadi) [reply src/Workspaces/Core/Desktop/Workspace/SQLite/SQLitePersistentStorage_StringIds.cs]: No.  Issue was leaking out a "busy" exception.

**Pull request merged**